src/lib/comms/sol-lwm2m-common.c

```diff
@@ -0,0 +1,976 @@
+/*
+ * This file is part of the Soletta (TM) Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include <float.h>
+#include <stdlib.h>
+#include <time.h>
+#include <math.h>
+#include <ctype.h>
+
+#define SOL_LOG_DOMAIN &_lwm2m_common_domain
+
+#include "sol-log-internal.h"
+#include "sol-util-internal.h"
+#include "sol-list.h"
+#include "sol-lwm2m.h"
+#include "sol-lwm2m-common.h"
+#include "sol-macros.h"
+#include "sol-mainloop.h"
+#include "sol-monitors.h"
+#include "sol-random.h"
+#include "sol-str-slice.h"
+#include "sol-str-table.h"
+#include "sol-util.h"
+#include "sol-http.h"
+
+SOL_LOG_INTERNAL_DECLARE_STATIC(_lwm2m_common_domain, "lwm2m-common");
+
+int sol_lwm2m_common_init(void);
+void sol_lwm2m_common_shutdown(void);
```

Comment:
no, that's the way since we require forward definition for non-static functions (so people remember to add them to headers or to make them static). These are the exceptions and we're handling like you did.